geomesa-redis/geomesa-redis-datastore/src/main/scala/org/locationtech/geomesa/redis/data/RedisDataStoreFactory.scala

```diff
@@ -85,51 +94,86 @@ object RedisDataStoreFactory extends GeoMesaDataStoreInfo with LazyLogging {
+            // else use the redis:// protocol to support databases, etc
+            val parsed =
+              if (url.indexOf(":") == -1) {
+                Try(new URI(url))
+              } else {
+                parse(url).orElse(parse(s"redis://$url"))
+              }
+            parsed match {
+              case Success(uri) => uri
+              case Failure(e) =>
+                throw new IllegalArgumentException(s"Could not create valid Redis connection URI from: $url", e)
+            }
+          }
+          val config = new GenericObjectPoolConfig[Jedis]()
+          PoolSizeParam.lookupOpt(params).foreach(s => config.setMaxTotal(s.intValue()))
+          config.setTestOnBorrow(TestConnectionParam.lookup(params))
+          val timeout = SocketTimeoutParam.lookup(params).toMillis.toInt
+
+          new JedisPool(config, uri, timeout)
+        case true =>
+          val urls = RedisUrlParam.lookup(params).split(",")
+          if (urls.isEmpty) {
+            throw new IllegalArgumentException("No Redis URLs provided. Please set 'redis.url'")
           }
-        parsed match {
-          case Success(uri) => uri
-          case Failure(e) =>
-            throw new IllegalArgumentException(s"Could not create valid Redis connection URI from: $url", e)
-        }
+          val config = new GenericObjectPoolConfig[JedisCluster]()
+          PoolSizeParam.lookupOpt(params).foreach(s => config.setMaxTotal(s.intValue()))
+          config.setTestOnBorrow(TestConnectionParam.lookup(params))
+          val timeout = SocketTimeoutParam.lookup(params).toMillis.toInt
+
+          val clusterNodes: java.util.Set[HostAndPort] =
+            urls.map { url =>
+              val parsed = parse(url).getOrElse {
+                throw new IllegalArgumentException(s"Could not create valid Redis connection URI from: $url")
+              }
+              JedisURIHelper.getHostAndPort(parsed)
+            }.toSet.asJava
+          val jedisClientConfig: JedisClientConfig =
+            DefaultJedisClientConfig.builder()
+              .timeoutMillis(timeout)
+              .build()
+          val connectionPoolConfig: GenericObjectPoolConfig[Connection] = new GenericObjectPoolConfig[Connection]()
+          val clusterFactory = new JedisClusterFactory(clusterNodes, jedisClientConfig, connectionPoolConfig)
+          new Pool[JedisCluster](config, clusterFactory)
       }
-
-      val config = new GenericObjectPoolConfig[Jedis]()
-      PoolSizeParam.lookupOpt(params).foreach(s => config.setMaxTotal(s.intValue()))
-      config.setTestOnBorrow(TestConnectionParam.lookup(params))
-      val timeout = SocketTimeoutParam.lookup(params).toMillis.toInt
-
-      new JedisPool(config, uri, timeout)
     }
   }
 
   /**
-    * Builds configuration from data store parameters
-    *
-    * @param params params
-    * @return
-    */
+   * Builds configuration from data store parameters
+   *
+   * @param params params
+   * @return
+   */
```

Comment:
i think all the changes from here and below are just formatting changes, could you revert them to minimize the diff?